plugins/group_ownership_policy.py

```diff
@@ -0,0 +1,73 @@
+from sqlalchemy.sql import label
+
+from grouper.models.base.constants import OBJ_TYPES
+from grouper.models.group import Group
+from grouper.models.group_edge import GROUP_EDGE_ROLES, OWNER_ROLE_INDICES, GroupEdge
+from grouper.models.user import User
+from grouper.plugin import BasePlugin
+from grouper.user_group import get_groups_by_user
+
+
+class GroupOwnershipPolicyViolation(Exception):
+    """This exception is raised when trying to remove the last owner of a group."""
+    pass
```

Comment:
instead of making this a plugin implementation, can we move this to `grouper.plugin` and 'require' plugins to throw this exact exception? I would make it so the calling code doesn't have to catch a blanket `Exception`.

unclear if that's too awkward though. would like to hear your thoughts on it.

Reply:
yeah, I'm not terribly happy about catching blanket `Exception`s either, but it didn't feel right to pull in a plugin-specific exception either. perhaps we can have a generic `PluginException` or something along those lines?